File: Server/src/commonStructures/networkOwnedHexagonList.ts
import Position from "./position";

/**
 * A class that contains the information about hexagons owned by a player
 */
export default class NetworkOwnedHexagonList {
    /**
     * The id of the player
     */
    public id:number;
    /**
     * The list of hexagons owned by the player
     */
    public hexagonList:Position[];
    /**
     * Create a new instance of NetworkOwnedHexagonList
     * @param id The id of the player
     * @param hexagonList The list of hexagons owned by the player
     */
    public constructor(id:number, hexagonList:Position[]) {
        this.id = id;
        this.hexagonList = hexagonList;
    }
}

File: Assets/Scripts/Network/CommonStructures/NetworkOwnedHexagonList.cs
namespace Network
{
    /// <summary>
    /// A class that contains the information about hexagons owned by a player
    /// </summary>
    public class NetworkOwnedHexagonList
    {
        /// <summary>
        /// The id of the player 
        /// </summary>
        public int id;
        /// <summary>
        /// The list of hexagons owned by the player
        /// </summary>
        public Position[] hexagonList;
        /// <summary>
        /// Create a new instance of NetworkOwnedHexagonList
        /// </summary>
        /// <param name="id">The id of the player</param>
        /// <param name="hexagonList">The list of hexagons owned by the player</param>
        public NetworkOwnedHexagonList(int id,Position[] hexagonList)
        {
            this.id = id;
            this.hexagonList = hexagonList;
        }
    }
}
